services/wfm/src/main/java/org/openkilda/wfm/topology/portstate/spout/SwitchPortsSpout.java

```diff
@@ -65,23 +66,19 @@ public void open(Map map, TopologyContext context, SpoutOutputCollector collecto
 
     @Override
     public void nextTuple() {
-        Message message = buildPortsCommand(REQUESTER);
-        logger.debug("emitting portsCommand: {}", message.toString());
-
-        try {
-            Values values = new Values(PAYLOAD, Utils.MAPPER.writeValueAsString(message));
-            collector.emit(values);
-        } catch (JsonProcessingException e) {
-            logger.error("Error sleeping");
-        }
-
-        // Note that no tupleId which means this is an untracked tuple which is
-        // required for the sleep
-        try {
-            Thread.sleep(frequency * 1000);
-        } catch (InterruptedException e) {
-            logger.error("Error sleeping");
+        if (now() - lastTickTime > frequency) {
+            Message message = buildPortsCommand(REQUESTER);
+            logger.debug("emitting portsCommand: {}", message.toString());
+            try {
+                Values values = new Values(PAYLOAD, Utils.MAPPER.writeValueAsString(message));
```

Comment:
Due to constant placed in kafka's "key" field - all produced records will be placed in one/same topic-partition.

Reply:
done